src/cpp/src/text2image/stable_diffusion_xl_pipeline.hpp

```diff
@@ -0,0 +1,345 @@
+        for (size_t inference_step = 0; inference_step < generation_config.num_inference_steps; inference_step++) {
+            // concat the same latent twice along a batch dimension in case of CFG
+            if (batch_size_multiplier > 1) {
+                batch_copy(latent, latent_cfg, 0, 0, generation_config.num_images_per_prompt);
+                batch_copy(latent, latent_cfg, 0, generation_config.num_images_per_prompt, generation_config.num_images_per_prompt);
+            } else {
+                // just assign to save memory copy
+                latent_cfg = latent;
+            }
+
+            m_scheduler->scale_model_input(latent_cfg, inference_step);
+
+            ov::Tensor timestep(ov::element::i64, {1}, &timesteps[inference_step]);
+            ov::Tensor noise_pred_tensor = m_unet->infer(latent_cfg, timestep);
+
+            ov::Shape noise_pred_shape = noise_pred_tensor.get_shape();
+            noise_pred_shape[0] /= batch_size_multiplier;
+            noisy_residual_tensor.set_shape(noise_pred_shape);
+
+            if (batch_size_multiplier > 1) {
+                // perform guidance
+                float* noisy_residual = noisy_residual_tensor.data<float>();
+                const float* noise_pred_uncond = noise_pred_tensor.data<const float>();
+                const float* noise_pred_text = noise_pred_uncond + noisy_residual_tensor.get_size();
+
+                for (size_t i = 0; i < noisy_residual_tensor.get_size(); ++i) {
+                    noisy_residual[i] = noise_pred_uncond[i] +
+                        generation_config.guidance_scale * (noise_pred_text[i] - noise_pred_uncond[i]);
+                }
+            } else {
+                noisy_residual_tensor = noise_pred_tensor;
+            }
+
+            auto scheduler_step_result = m_scheduler->step(noisy_residual_tensor, latent, inference_step);
+            latent = scheduler_step_result["latent"];
+
+            // check whether scheduler returns "denoised" image, which should be passed to VAE decoder
+            const auto it = scheduler_step_result.find("denoised");
+            denoised = it != scheduler_step_result.end() ? it->second : latent;
+        }
+
+        return m_vae_decoder->infer(denoised);
+    }
+
+private:
+    bool do_classifier_free_guidance(float guidance_scale) const {
+        return guidance_scale > 1.0 && m_unet->get_config().time_cond_proj_dim < 0;
+    }
+
+    void initialize_generation_config(const std::string& class_name) override {
+        assert(m_unet != nullptr);
+        const auto& unet_config = m_unet->get_config();
+        const size_t vae_scale_factor = m_unet->get_vae_scale_factor();
+
+        m_generation_config.height = unet_config.sample_size * vae_scale_factor;
+        m_generation_config.width = unet_config.sample_size * vae_scale_factor;
+
+        if (class_name == "StableDiffusionXLPipeline") {
+            m_generation_config.guidance_scale = 7.5f;
```

Comment:
I see that default value of guidance scale for SDXL is 5.0 https://github.com/huggingface/diffusers/blob/v0.30.3/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L831